src/v/kafka/server/handlers/alter_configs.cc

```diff
@@ -81,6 +81,17 @@ void parse_and_set_tristate(
     }
 }
 
+static void parse_and_set_topic_replication_factor(
+  cluster::property_update<std::optional<cluster::replication_factor>>&
+    property,
+  const std::optional<ss::sstring>& value) {
+    if (!value) {
+        property.value = std::nullopt;
+    }
+
+    property.value = cluster::parsing_replication_factor(value);
```

Comment:
this method will cause undefined behavior in cases where `value` is `std::nullopt` since control falls through to `parsing_replicat_factor` which unconditionally deferences `value`.

cc @jcsp @mmaslankaprv